incubator/declarative-linking/src/main/java/org/glassfish/jersey/linking/EntityDescriptor.java

```diff
@@ -190,4 +212,44 @@ private void findLinkHeaders(Class<?> entityClass) {
             findLinkHeaders(ic);
         }
     }
+    
+    /**
+     * Find and cache the methods of the supplied class and its superclasses and
+     * interfaces.
+     * @param entityClass the class
+     */
+    private void findMethods(Class<?> entityClass) {
+        for (Method m: entityClass.getMethods()) {
+            InjectLink a = m.getAnnotation(InjectLink.class);
+            Class<?> t = m.getReturnType();
+            if (a != null) {
+                if (t.equals(String.class) || t.equals(URI.class) || Link.class.isAssignableFrom(t)) {
+                    if (!linkMethods.containsKey(m.getName())) {
+                    	linkMethods.put(m.getName(), new InjectLinkMethodDescriptor(m, a, t));
+                    }
+                }  else {
+                    // TODO unsupported type
+                }
+            } else if (m.isAnnotationPresent(InjectLinks.class)) {
+                
+                if (List.class.isAssignableFrom(t)
+                        || t.isArray() && Link.class.isAssignableFrom(t.getComponentType())) {
+                    
+                    InjectLinks a2 = m.getAnnotation(InjectLinks.class);
+                    linkMethods.put(m.getName(), new InjectLinksMethodDescriptor(m, a2, t));
+                } else {
+                    throw new IllegalArgumentException("Can only inject links onto a List<Link> or Link[] object");
```

Comment:
If you decide to leave the method as it is, than it might make sense to "customize" the message to differentiate the message from the one with field injection.